types/passport/index.d.ts

```diff
@@ -29,8 +29,10 @@ declare global {
             logIn(user: User, done: (err: any) => void): void;
             logIn(user: User, options: any, done: (err: any) => void): void;
 
-            logout(): void;
-            logOut(): void;
+            logout(options: { keepSessionInfo?: boolean }, done: (err: any) => void): void;
+            logout(done: (err: any) => void): void;
+            logOut(options: { keepSessionInfo?: boolean }, done: (err: any) => void): void;
+            logOut(done: (err: any) => void): void;
```

Comment:
These callbacks appear to be optional.

```suggestion
            logout(options: { keepSessionInfo?: boolean }, done?: (err: any) => void): void;
            logout(done?: (err: any) => void): void;
            logOut(options: { keepSessionInfo?: boolean }, done?: (err: any) => void): void;
            logOut(done?: (err: any) => void): void;
```

Reply:
Not providing a callback function appears to lead to the following error:
```
Error: req#logout requires a callback function
    at req.logout.req.logOut (/.../node_modules/passport/lib/http/request.js:65:44)
    at /.../dist/app.js:11:9
    at Layer.handle [as handle_request] (/.../node_modules/express/lib/router/layer.js:95:5)
    at next (/.../node_modules/express/lib/router/route.js:144:13)
    at Route.dispatch (/.../node_modules/express/lib/router/route.js:114:3)
    at Layer.handle [as handle_request] (/.../node_modules/express/lib/router/layer.js:95:5)
    at /.../node_modules/express/lib/router/index.js:284:15
    at Function.process_params (/.../node_modules/express/lib/router/index.js:346:12)
    at next (/.../node_modules/express/lib/router/index.js:280:10)
    at initialize (/.../node_modules/passport/lib/middleware/initialize.js:98:5)
```

```typescript
import express from 'express';
import passport from 'passport';
const app = express()

app.use(passport.initialize())

app.post('/logout', (req, res, next) => {
    // @ts-ignore
    req.logout({ keepSessionInfo: false });
})
app.listen(3000)
```

Reply:
It looks like that's only the case if there's a session manager defined, so it might be possible to use Passport without that?

https://github.com/jaredhanson/passport/blob/42630cbd1ffd44d146ff96f0a4be6f3c12f81d75/lib/http/request.js#L64-L70

Reply:
I asked for clarification on https://github.com/jaredhanson/passport/discussions/901.

Reply:
I think it's better to keep as is for now, and make this change in the future.